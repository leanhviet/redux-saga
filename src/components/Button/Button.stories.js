// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import Button from './index'

storiesOf('Button', module)
  .add('Default', () => (
    <Button label="default" handleOnclick={action('clicked')} />
  ))
  .add('Disabled', () => (
    <Button label="default" handleOnclick={action('clicked')} disabled />
  ))
