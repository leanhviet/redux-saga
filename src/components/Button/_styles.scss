.button {
  background-color: $red;
  color: $white;
  padding: 17px 25px;
  text-transform: uppercase;
  font-weight: 700;
  font-size: 12px;
  border: 2px solid $red;
  float: left;
  border-radius: 999px;

  &:hover {
    @include transition(all 0.3s ease-in-out);

    background-color: $white;
    color: $red;
    border: 2px solid $red;
    text-decoration: none;
  }

  &.success {
    color: #004085;
    background-color: #cce5ff;
    border-color: #b8daff;
  }
}
