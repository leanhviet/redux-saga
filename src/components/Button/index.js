// @flow
import React from 'react'

type Props = {
  label: string,
  handleOnclick: Function,
  className: string,
  disabled: boolean
}

const Button = (props: Props) => {
  const { label, className, handleOnclick, disabled } = props

  return (
    <button
      className={`button ${className}`}
      onClick={handleOnclick}
      disabled={disabled}
    >
      {label}
    </button>
  )
}

Button.defaultProps = {
  label: '',
  handleOnclick: () => {},
  className: '',
  disabled: false
}

export default Button
