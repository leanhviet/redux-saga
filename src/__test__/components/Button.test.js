// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import Button from '../../components/Button'

const mockClickFunc = jest.fn()
const mockProps = {
  label: 'Search',
  handleOnclick: mockClickFunc,
  className: 'mockClass',
  disabled: false
}

const wrapper = (props = {}) => {
  const component = shallow(<Button {...props} />)

  return component
}

describe('Button component', () => {
  const wrapperWithoutProps = wrapper()
  const wrapperWithProps = wrapper(mockProps)

  // Snapshot
  test('should render correctly Button component default props', () => {
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
  })

  // Testing props
  describe('Handle with props', () => {
    test('should is rendered default props', () => {
      expect(wrapperWithoutProps.find('button').props().children).toBeFalsy()
    })

    test('should is rendered with a custom Props', () => {
      expect(wrapperWithProps.find('button').props().children).toMatch(
        mockProps.label
      )

      expect(wrapperWithProps.find('button').props().className).toMatch(
        mockProps.className
      )

      expect(wrapperWithProps.find('button').props().disabled).toBe(
        mockProps.disabled
      )
    })
  })

  // Testing events
  describe('Checking events', () => {
    test('check the on Blur callback ', () => {
      wrapperWithProps.simulate('click')

      expect(mockClickFunc).toHaveBeenCalledTimes(1)
    })
  })
})
