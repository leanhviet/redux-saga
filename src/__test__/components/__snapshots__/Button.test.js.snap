// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly Button component default props 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
/>
`;
